container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -58,13 +73,82 @@ if [[ ! -z ${TEMPESTCONF_REMOVE} ]]; then
     done <<< "$TEMPESTCONF_REMOVE"
 fi
 
-TEMPESTCONF_OVERRIDES="$(echo ${TEMPESTCONF_OVERRIDES} | tr '\n' ' ') identity.v3_endpoint_type public"
 
 if [ -n "$CONCURRENCY" ] && [ -z ${TEMPEST_CONCURRENCY} ]; then
     TEMPEST_ARGS+="--concurrency ${CONCURRENCY} "
 fi
 
-pushd $HOMEDIR
+function run_git_tempest {
+    mkdir -p $TEMPEST_EXTERNAL_PLUGIN_DIR
+    pushd $TEMPEST_EXTERNAL_PLUGIN_DIR
+
+    python3 -m venv .venv
+    source ./.venv/bin/activate
+
+    for plugin_index in "${!TEMPEST_EXTERNAL_PLUGIN_GIT_URL[@]}"; do
+        git_url=${TEMPEST_EXTERNAL_PLUGIN_GIT_URL[plugin_index]}
+        change_url=${TEMPEST_EXTERNAL_PLUGIN_CHANGE_URL[plugin_index]}
+        refspec=${TEMPEST_EXTERNAL_PLUGIN_REFSPEC[plugin_index]}
+        plugin_name=$(basename -s .git $git_url)
+
+        git clone $git_url
+
+        if [[ ! -z $change_url ]] && [[ ! -z $refspec ]] || \
+           [[ $change_url != "-" ]] && [[ $refspec != "-" ]]; then
```

Comment:
maybe a comment would help but i'm trying to imagine what $change_url looks like here and why we need to guard for != "-" (and failing to imagine it :))

Reply:
I explained the usage of the variables used here in the comment at the beginning of the script. I can imagine that the "-" is a little bit confusing. I think that it would be used only in edge cases when someone would want to download multiple changes.

I think that the conversation here https://github.com/openstack-k8s-operators/test-operator/pull/12 can be helpful as well.

Reply:
I got it. Not sure if there is a better way to implement it, but I see that you need "-" because your loop gets values based on indexes.
A good documentation was added to the script itself. Is there another place that we can add this? So everybody would know how to use this?